### PR TITLE
Connect Jenkins agent to LUMI slots

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Node is launched with `.cms-bot/lumi/connect.sh $OS cms-bot/lumi/jenkins_java.sh` with `OS` value being `el8` or `el9`.

Script `update_image.sh` runs on the lumi user node and updates the `.sif` images given a target OS and architecture.

### CODE REVIEW

File: lumi/connect.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
+
+echo "Launching LUMI node..."
+
+JENKINS_VERSION=$(jenkins --version)
+OS=$1
+SCRIPT=$2
+
+scp -q -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ServerAliveInterval=60 /var/lib/jenkins/slave.jar andbocci@lumi.csc.fi:~/slave-${JENKINS_VERSION}.jar
+ssh -q -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ServerAliveInterval=60 andbocci@lumi.csc.fi "./get_slot.sh ${JENKINS_VERSION} ${OS} ${SCRIPT}"
```
[the remark this answers is not in the document]
Comment (aandvalenzuela) [reply lumi/connect.sh:10]: I am defining `SSH_OPTS=` too

[the remark this answers is not in the document]
Comment (smuzaffar) [reply lumi/connect.sh:10]: also `jenkins_java.sh` is not available on user node. may be better to scp `cms-bot/lumi` to user node

File: lumi/get_slot.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -ex
+
+echo "Requesting slot on LUMI HPC..."
+JENKINS_VERSION=$1
+echo "Jenkins version is $JENKINS_VERSION"
```
Comment (smuzaffar) [inline lumi/get_slot.sh:5]: No need to pass jenkins version here.  you can run the script e.g.
```
~/jenkins_lumi/get_slot.sh <OS> ~/jenkins_lumi/jenkins_java.sh ~/jenkins_lumi/slave-${JENKINS_VERSION}.jar`
```

File: lumi/jenkins_java.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -ex
+
+/etc/alternatives/jre_11/bin/java -version # java > 11 needed for Jenkins connection
+/etc/alternatives/jre_11/bin/java --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.reflect=ALL-UNNAMED -jar /users/andbocci/slave.jar -jar-cache /users/andbocci/tmp
```
Comment (smuzaffar) [inline lumi/jenkins_java.sh:4]: why ` /users/andbocci/slave.jar` ? it should be with jenkins version

File: lumi/get_slot.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+OS=$2
+SCRIPT=$3
```
Comment (smuzaffar) [inline lumi/get_slot.sh:8]: do we need to pass the script name here? isn't it going to be `$(dirname $0)/jenkins_java.sh` always?

Comment (aandvalenzuela) [reply lumi/get_slot.sh:8]: Yes, you are is right, thanks!

**Comment by aandvalenzuela** [comment]
Converting it to draft to create different workspaces for different nodes (e.g., working with `el8` and `el9` at the same time)

**Pull request merged**